### ISSUE TITLE
false positive RedundantCondition for static::class in trait?

### ISSUE THREAD

**Issue by ostrolucky** [vimeo/psalm#715]
```php
<?php
  
trait foo {
  public function bar() {
    $type = static::class;
    $r = new \ReflectionClass($type);
    $values = $r->getConstants();
    $callback = function ($v) {
        return \is_int($v) || \is_string($v);
      };
    
    if (is_a($type, \stdClass::class, true)) {
      $callback = function ($v) {
        return \is_int($v) && 0 === ($v & $v - 1) && $v > 0;
      };
    }
    
    return array_filter($values, $callback);
  }
}

class bar {
  use foo;
}
```
> ERROR: RedundantCondition - 12:9 - Found a redundant condition when evaluating $type and trying to reconcile type 'class-string' to stdClass

https://getpsalm.org/r/bc4e0c4892

**Comment by muglug** [comment]
Thanks!
